packages/cln_plugin/lib/src/cln_plugin_base.dart

```diff
@@ -86,8 +89,12 @@ class Plugin implements CLNPlugin {
   }
 
   @override
-  void registerHook({required String name}) {
+  void registerHook(
+      {required String name,
+      required Future<Map<String, Object>> Function(Plugin, Map<String, Object>)
```

Comment:
```suggestion
      {required String name,
      List<String>? before,
      List<String>? after,
      required Future<Map<String, Object>> Function(Plugin, Map<String, Object>)
```

If I remember correct we have also the before and after methods rules right? :) maybe you was thinking to add after! BTW I add a coment